Add prune API page for 5.19 (#2341)
sensu/sensu-docs: content/sensu-go/5.19/api/prune.md

---
title: "Prune API"
description: "The Sensu prune API provides HTTP access to create pruning commands to delete resources that do not appear in a given set of Sensu objects from a file, URL, or STDIN. This reference includes an example for creating a prune command for your Sensu instance. Read on for the full reference."
version: "5.19"
product: "Sensu Go"
menu:
  sensu-go-5.19:
    parent: api
---

## The `/prune/v1alpha` API endpoint

{{% notice important %}}
**IMPORTANT**: The prune API is an alpha feature in release 5.19.0 and may include breaking changes.
{{% /notice %}}

**COMMERCIAL FEATURE**: Access sensuctl pruning in the packaged Sensu Go distribution. For more information, see [Get started with commercial features][1].

### `/prune/v1alpha` (POST)

> The `/prune/v1alpha` API endpoint provides HTTP POST access to create a pruning command to delete resources that do not appear in a given set of Sensu objects from a file, URL, or STDIN.

Reviewer: > in a given set of Sensu objects from a file, URL, or STDIN.

That might be a bit misleading in the context of the API, since users will have to provide the Sensu objects themselves to curl. WDYT?

Author: I agree -- I copied that from sensuctl and then forgot to update it to match the API. Thank you.

> The `/prune/v1alpha` API endpoint provides HTTP POST access to create a pruning command to delete resources that do not appear in a given set of Sensu objects from a file, URL, or STDIN.

Author: ```suggestion
The `/prune/v1alpha` API endpoint provides HTTP POST access to create a pruning command to delete resources that are not specified in the request body.
```

-H "Authorization: Bearer $SENSU_ACCESS_TOKEN" \
-H 'Content-Type: application/json' \
-d '{
  "type": "CheckConfig",
  "api_version": "core/v2",
  "name": "check-echo",
  "namespace": "dev",
  "labels": {
    "region": "us-west-2",
    "sensu.io/managed_by": "sensuctl"
  },
  "created_by": "admin"
}'

HTTP/1.1 201 Created

[
  {
    "type": "CheckConfig",
    "api_version": "core/v2",
    "name": "check-echo",
    "namespace": "dev",
    "labels": {
      "region": "us-west-2",
      "sensu.io/managed_by": "sensuctl"
    },
    "created_by": "admin"
  }
]
{{< /highlight >}}

#### API Specification {#prune-v1alpha-specification}

/prune/v1alpha (POST) | 
----------------------|------
description           | Creates a pruning command to delete the specified resources.
example URL           | http://hostname:8080/api/enterprise/prune/v1alpha
example payload       | {{< highlight shell >}}
{
  "type": "CheckConfig",
  "api_version": "core/v2",
  "name": "check-echo",
  "namespace": "dev",
  "labels": {
    "region": "us-west-2",
    "sensu.io/managed_by": "sensuctl"
  },
  "created_by": "admin"
}
{{< /highlight >}}

> query parameters      | <ul><li>`type`: The [fully-qualified name][2] of the resource you want to prune. Example: `?type=core/v2.CheckConfig`.</li><li>`allUsers`: Prune resources created by all users. Mutually exclusive with the `users` parameter. Defaults to false. Example: `?allUsers=true`.</li><li>`clusterWide`: Prune any cluster-wide (non-namespaced) resources that are not defined in the configuration. Defaults to false. Example: `?clusterWide=true`.</li><li>`dryRun`: Print the resources that will be pruned but does not actually delete them. Defaults to false. Example: `?dryRun=true`.</li><li>`labelSelector`: Prune only resources that match the specified labels (accepts multiple values). Labels are a [commercial feature][1]. Example: `?labelSelector=[...]`.</li><li>`namespaces`: The namespace where you want to apply pruning. Example: `?namespaces=dev`.</li><li>`users`: Prune only resources that were created by the specified users (accepts multiple values). Defaults to the currently configured sensuctl user. Example: `?users=admin`.</li></ul> To use multiple values for the parameters that allow them, you must specify the parameter multiple times (for example, `?users=admin&users=dev`) rather than using a comma-separated list.

Reviewer: Maybe it's just because of my _hugo_ version but this bit is kind of hard to read:
![Screen Shot 2020-04-07 at 10 39 18 AM](https://user-images.githubusercontent.com/4105580/78682538-1afa0880-78bc-11ea-8a6a-bd3835f9a972.png)

Author: Hm -- it should be a bulleted list. Could you try the preview at https://sensu-docs-prune-api-3aikztpiw.herokuapp.com/sensu-go/latest/api/prune/#prune-v1alpha-specification? That is what it would look like when published.

Reviewer: Okay yeah, I might have a too recent Hugo version or something missing, so ignore this then!
